File: src/com/algorithm/Main2884.java
package com.algorithm;

import java.util.Scanner;

public class Main2884 {
    public static void main(String[] args){

        int hour, minute, newMinute;

        Scanner input = new Scanner(System.in);

        hour = input.nextInt();
        minute = input.nextInt();

        if(minute-45<0){
            if(hour==0){
                hour=23;
            }else{
                hour -= 1;
            }
            newMinute = 59-(44-minute);
        }else{
            newMinute = minute-45;
        }

        System.out.println(hour+" "+newMinute);

    }
}

File: src/com/algorithmStudy/acmicpc/Main2884.py
s1, s2 = input().split()
hour = int(s1)
minute = int(s2)
if minute>=45:
    print(hour,minute-45)
else:
    if hour==0:
        print("23",minute+15)
    else:
        print(hour-1,minute+15)
